src/Controller/LogsController.php

```diff
@@ -62,4 +68,104 @@ public function gc()
+        }
+
+        if ($this->request->is('post')) {
+            $searchData = $search->prepareData($this->request);
+            if ('' !== $id) {
+                $search->update($searchData, $id);
+            }
+
+            if ('' === $id) {
+                $id = $search->create($searchData);
+            }
+
+            list($plugin, $controller) = pluginSplit($model);
+
+            return $this->redirect(['plugin' => $plugin, 'controller' => $controller, 'action' => __FUNCTION__, $id]);
+        }
+
+        $entity = $search->get($id);
+
+        $searchData = json_decode($entity->get('content'), true);
+        if ($this->request->is('ajax')) {
+            $this->viewBuilder()->layout('ajax');
+            $queryData = $this->getSearchFieldsFromArray((array)$this->request->getQuery());
+            $response = $this->getAjaxViewVars($queryData, $table, $search);
+            $this->set('types', $this->DatabaseLogs->getTypes());
+            $this->set('data', $response);
+            $this->set('module', $this->loadModel()->getAlias());
+
+            $this->render('advance_search_result');
+
+            return;
+        }
+
+        $searchData = SearchValidator::validateData($table, $searchData['latest'], $this->Auth->user());
+
+        // reset should only be applied to current search id (url parameter)
+        // and NOT on newly pre-saved searches and that's we do the ajax
+        // request check above, to prevent resetting the pre-saved search.
+        $search->reset($entity);
+
+        $this->set('searchableFields', Utility::instance()->getSearchableFields($table, $this->Auth->user()));
+        $this->set('savedSearches', $searchTable->getSavedSearches([$this->Auth->user('id')], [$model]));
+        $this->set('model', $model);
+        $this->set('modelAlias', $this->loadModel()->getAlias());
+        $this->set('searchData', $searchData);
+        $this->set('savedSearch', $entity);
+        $this->set('preSaveId', $search->create($searchData));
+        // INFO: this is valid when a saved search was modified and the form was re-submitted
+        $this->set('isEditable', $searchTable->isEditable($entity));
+        $this->set('searchOptions', SearchOptions::get());
+        $this->set('associationLabels', Utility::instance()->getAssociationLabels($table));
+
+        $this->render($this->searchElement);
+    }
+
+    /**
+     * Remove unnecessary data from array
+     *
+     * @param mixed[] $data List of data
+     * @return mixed[]
+     */
+    protected function getSearchFieldsFromArray(array $data): array
+    {
+        $searchFieldNamesAllowed = [
+            'criteria',
+            'fields',
+            'aggregator',
+            'display_columns',
+            'sort_by_field',
+            'sort_by_order',
+            'group_by'
+        ];
+
+        $newData = [];
+        foreach ($searchFieldNamesAllowed as $fieldName) {
```

Comment:
@mitsios i believe this can be replaced with `array_intersect_key($data, $searchFieldNamesAllowed)`
```suggestion
        $searchFieldNamesAllowed = [
            'criteria' => null,
            'fields' => null,
            'aggregator' => null,
            'display_columns' => null,
            'sort_by_field' => null,
            'sort_by_order' => null,
            'group_by' => null,
        ];
        
        return array_intersect_key($data, $searchFieldNamesAllowed);
```